src/handlers/signup.rs

```diff
@@ -0,0 +1,30 @@
+use actix_web::web::{Data, Json};
+use actix_web::Responder;
+use serde::Deserialize;
+
+use crate::domain::services::dto::user::RegisterUserDto;
+use crate::state::State;
+
+#[derive(Debug, Deserialize)]
+pub struct SignupPayload {
+    pub email: String,
+    pub username: String,
+    pub name: String,
+    pub surname: Option<String>,
+    pub password: String,
+}
+
+pub async fn signup(state: Data<State>, payload: Json<SignupPayload>) -> impl Responder {
+    let payload = RegisterUserDto {
+        email: payload.email.clone(),
+        username: payload.username.clone(),
+        name: payload.name.clone(),
+        surname: payload.surname.clone(),
+        password: payload.password.clone(),
+    };
```

Comment:
These clones are awful but we can't move fields out of `payload` so.. I will check how can we improve this I'm not 100% sure if this is a common pattern for `Json` type.

Reply:
This was easily fixed using `RegisterUserDto` straight and `into_inner`!